snuba/datasets/storages/generic_metrics.py

```diff
@@ -0,0 +1,58 @@
+"""
+The storages defined in this file are for the generic metrics system,
+initially built to handle metrics-enhanced performance.
+"""
+from snuba.clickhouse.columns import (
+    AggregateFunction,
+    Array,
+    Column,
+    ColumnSet,
+    DateTime,
+    Nested,
+    SchemaModifiers,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.schemas.tables import TableSchema
+from snuba.datasets.storage import ReadableTableStorage
+from snuba.datasets.storages import StorageKey
+from snuba.query.processors.arrayjoin_keyvalue_optimizer import (
+    ArrayJoinKeyValueOptimizer,
+)
+from snuba.query.processors.table_rate_limit import TableRateLimit
+
+aggregated_columns = [
+    Column("org_id", UInt(64)),
+    Column("use_case_id", String()),
+    Column("project_id", UInt(64)),
+    Column("metric_id", UInt(64)),
+    Column("granularity", UInt(8)),
+    Column("timestamp", DateTime()),
+    Column("retention_days", UInt(16)),
+    Column("raw_tags", Nested([("key", UInt(64)), ("value", String())])),
+    Column("_raw_tags_hash", Array(UInt(64), SchemaModifiers(readonly=True))),
+    Column("indexed_tags", Nested([("key", UInt(64)), ("value", String())])),
```

Comment:
You should not need to store the tag keys twice. Can't we have a tag keys column and two tag values (one for strings and one for ints) ? 
Also there is an issue here. The indexed_tags value should be a `UInt64`.